wpilibc/src/main/native/cpp/DoubleSolenoid.cpp

```diff
@@ -130,6 +130,15 @@ DoubleSolenoid::Value DoubleSolenoid::Get() const {
   return kOff;
 }
 
+void DoubleSolenoid::Toggle() {
+  Value value = Get();
```

Comment:
Calling `Get()` results in a HAL call and can block for a relatively long period of time (we had a lot of loop overruns). Would it be better if you stored the state of the solenoid as a boolean member variable and used that to set instead?

This would also require setting the member variable's value whenever `Set()` is called.

Reply:
If it's safe to assume the solenoid's value won't change unless Set() is called, Get() never needs to call HAL_GetSolenoid(). If that isn't the case though, we won't be able to cache the value of Get().